sdk/scheduler/src/test/java/com/mesosphere/sdk/http/endpoints/HealthResourceTest.java

```diff
@@ -1,80 +1,1171 @@
-        resource = new HealthResource(Arrays.asList(
-                DefaultPlanManager.createProceeding(mockPlan1),
-                DefaultPlanManager.createProceeding(mockPlan2)), SchedulerConfigTestUtils.getTestSchedulerConfig());
-    }
+  private static final PodSpec helloPodSpec = TestPodFactory.getPodSpec(
+      TASK_HELLO_POD_NAME,
+      TestConstants.RESOURCE_SET_ID + "-hello",
+      TASK_HELLO_NAME,
+      TASK_HELLO_CMD,
+      TestConstants.SERVICE_USER,
+      TASK_COUNT,
+      TASK_CPU,
+      TASK_MEM,
+      TASK_DISK); 
 
-    @Test
-    public void testPlanErrorIncomplete() {
-        when(mockPlan1.getErrors()).thenReturn(Collections.singletonList("err"));
-        when(mockPlan1.isComplete()).thenReturn(true);
-        when(mockPlan2.getErrors()).thenReturn(Collections.emptyList());
-        when(mockPlan2.isComplete()).thenReturn(false);
-        checkHealthStatus(Response.Status.EXPECTATION_FAILED);
-    }
+  private static final PodSpec worldPodSpec = TestPodFactory.getPodSpec(
+      TASK_WORLD_POD_NAME,
+      TestConstants.RESOURCE_SET_ID + "-world",
+      TASK_WORLD_NAME,
+      TASK_WORLD_CMD,
+      TestConstants.SERVICE_USER,
+      TASK_COUNT,
+      TASK_CPU,
+      TASK_MEM,
+      TASK_DISK); 
 
-    @Test
-    public void testPlanError() {
-        when(mockPlan1.getErrors()).thenReturn(Collections.singletonList("err"));
-        when(mockPlan1.isComplete()).thenReturn(true);
-        when(mockPlan2.getErrors()).thenReturn(Collections.emptyList());
-        when(mockPlan2.isComplete()).thenReturn(true);
-        checkHealthStatus(Response.Status.EXPECTATION_FAILED);
-    }
+  private PodInstanceRequirement helloPodInstanceRequirement = getPodInstanceRequirement(helloPodSpec, 0);
+  private PodInstanceRequirement worldPodInstanceRequirement = getPodInstanceRequirement(worldPodSpec, 0);
 
-    @Test
-    public void testPlanIncomplete() {
-        when(mockPlan1.getErrors()).thenReturn(Collections.emptyList());
-        when(mockPlan1.isComplete()).thenReturn(false);
-        when(mockPlan2.getErrors()).thenReturn(Collections.emptyList());
-        when(mockPlan2.isComplete()).thenReturn(true);
-        checkHealthStatus(Response.Status.ACCEPTED);
-    }
+  private PodInstanceRequirement getPodInstanceRequirement(PodSpec podSpec, int index) {
+    PodInstance podInstance = new DefaultPodInstance(podSpec, index);
+    return PodInstanceRequirement.newBuilder(
+        podInstance,
+        podSpec.getTasks().stream().map(TaskSpec::getName).collect(Collectors.toList()))
+        .build();
+  }
 
-    @Test
-    public void testPlanComplete() {
-        when(mockPlan1.getErrors()).thenReturn(Collections.emptyList());
-        when(mockPlan1.isComplete()).thenReturn(true);
-        when(mockPlan2.getErrors()).thenReturn(Collections.emptyList());
-        when(mockPlan2.isComplete()).thenReturn(true);
-        checkHealthStatus(Response.Status.OK);
+  @Before
+  public void setupTest() throws Exception {
```

Comment:
wow mockito makes this `throws Exception` ?!

Reply:
good catch! fixed